### PR TITLE
cortexso 0.5.0-37

### PR DESCRIPTION
**Pull request by hiento09**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by branchv — CHANGES_REQUESTED** [review]
Thanks for the PR! A few comments

File: Formula/c/cortexso.rb [hunk 1]
```diff
@@ -17,12 +17,21 @@ class Cortexso < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f02d924016dcab7dce2faecaac5a77ba04ad4ba20dcac3e0fcc753d248364264"
   end
 
-  depends_on "node"
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "node@20"
+  depends_on "yarn"
```
Comment (branchvincent) [inline Formula/c/cortexso.rb:24]:
```suggestion
  depends_on "yarn" => :build
```

File: Formula/c/cortexso.rb [hunk 2]
```diff
@@ -17,12 +17,21 @@ class Cortexso < Formula
...
+
+  on_macos do
+    depends_on xcode: :build
```
[the remark this answers is not in the document]
Comment (hiento09) [reply Formula/c/cortexso.rb:27]: Hi, on my machine, if I remove Xcode, the build fails, but if I include Xcode, the build succeeds.
Command: `brew install --build-from-source ./Formula/c/cortexso.rb --verbose`

File: Formula/c/cortexso.rb [hunk 3]
```diff
@@ -38,14 +47,27 @@ def install
     node_modules.each_child do |dir|
       rm_r(dir) if dir.basename.to_s != "#{os}-#{arch}"
     end
+    # Remove build dependencies
```
[the remark this answers is not in the document]
Comment (hiento09) [reply Formula/c/cortexso.rb:51]: Got you, I am handling it in our `package.json` now, thank you for your suggestion.

File: Formula/c/cortexso.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (branchvincent) [reply Formula/c/cortexso.rb:27]: What's the error that you see? We can try removing it and CI will let us know

[the remark this answers is not in the document]
Comment (hiento09) [reply Formula/c/cortexso.rb:27]: Ok, let me try to remove and run it on CI now

[the remark this answers is not in the document]
Comment (hiento09) [reply Formula/c/cortexso.rb]: https://github.com/Homebrew/homebrew-core/actions/runs/10398092233/job/28794944122?pr=180461 you can check the error here @branchvincent , I got similar error locally

File: Formula/c/cortexso.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (hiento09) [reply Formula/c/cortexso.rb:47]: Hey @branchvincent.
Sorry for the inconvenience, I have bumped the version of our package to resolve this issue and renamed the branch while creating a new PR. Please review the new PR here https://github.com/Homebrew/homebrew-core/pull/181250

**Comment by chenrui333** [comment]
```
  npm info run cortex-cpp@0.5.0-36 install { code: 1, signal: null }
  npm verbose stack Error: command failed
  npm verbose stack     at promiseSpawn (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:22:22)
  npm verbose stack     at spawnWithShell (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:124:10)
  npm verbose stack     at promiseSpawn (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:12:12)
  npm verbose stack     at runScriptPkg (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script-pkg.js:77:13)
  npm verbose stack     at runScript (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/run-script.js:9:12)
  npm verbose stack     at /opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/rebuild.js:329:17
  npm verbose stack     at run (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:67:22)
  npm verbose stack     at /opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:84:9
  npm verbose stack     at new Promise (<anonymous>)
  npm verbose stack     at callLimit (/opt/homebrew/Cellar/node/22.6.0/libexec/lib/node_modules/npm/node_modules/promise-call-limit/dist/commonjs/index.js:35:69)
  npm verbose pkgid cortex-cpp@0.5.0-36
  npm error code 1
  npm error path /opt/homebrew/Cellar/cortexso/0.5.0-37/libexec/lib/node_modules/cortexso/node_modules/cortex-cpp
  npm error command failed
  npm error command sh -c prebuild-install --runtime napi --backend cmake-js --config Release || yarn rebuild
```

**Pull request closed without merging**